docs/src/manual/performance_tips.md

```diff
@@ -0,0 +1,47 @@
+
+## Faster problem formulation: Avoid scalar indexing
+
+When manipulating Convex.jl expressions, try to use a "vectorized" style, and avoid loops and scalar indexing. For example:
+
+```julia
+function bad_my_dot(x, y) # avoid this kind of code for Convex.jl expressions!
+    s = 0.0
+    for i in eachindex(x, y)
+        s += x[i]*y[i]
+    end
+    return s
+end
+```
+
+is usually perfectly reasonable Julia code (although perhaps `s` should be initialized as `s=zero(promote_type(eltype(x), eltype(y))))` to be more generic), since loops are typically fast in Julia.
+
+However, when using Convex.jl expressions like `x = Variable(100)`, calling `bad_my_dot(x, y)` with `y = rand(100)` will create 100 separate `IndexAtom`s, each of which has some overhead. That is because Convex maintains a lazy expression tree representing the problem. Instead, the vectorized form `sum(x .* y)` or `LinearAlgebra.dot(x, y)` will be more efficient.
+
+Indexing style typically should have no effect on solve times (step (3) above), only on constructing the expression tree and formulating the problem.
+
+## Speeding up solve times: dualization
+
+Depending on the solver and the problem, it can sometimes speed things up quite a lot to pass the solver the dual problem to solve rather than the primal problem. This can be accomplished easily with [Dualization.jl](https://github.com/jump-dev/Dualization.jl), simply by passing `dual_optimizer(optimizer)` instead of `optimizer` to `solve!`. See [Dualization](@ref) for an example.
+
+## Speeding up solve times: digging into the final formulation
+
+Convex.jl is built around conic programming, and it works with conic solvers. This may be a different optimization methodology than you may be used to, and it works by formulating problems in terms of affine objective functions and affine-function-in-cone constraints for a variety of convex cones. Convex.jl reformulates all problems to this form. For example, the objective function you give Convex will not actually be executed line-by-line; instead, it will be reformulated to a possibly very-different looking form for the conic solver to handle.
+
+Convex.jl provides an abstraction by automatically performing these reformulations, but as often is the case, performance leaks through. To have the best performance, you might need to learn more details about the various cones supported by your solver, by [MathOptInterface](https://jump.dev/MathOptInterface.jl/stable/reference/standard_form/#Vector-sets), and how Convex.jl reformulates your problem. There could be alternate mathematically-equivalent formulations that are more performant, and in some cases it could be that Convex's automatic reformulations can be improved.
```

Comment:
```suggestion
Convex.jl provides an abstraction by automatically performing these reformulations, but as often is the case, performance leaks through. To have the best performance, you might need to learn more details about the various cones supported by your solver, by [MathOptInterface](https://jump.dev/MathOptInterface.jl/stable/reference/standard_form/#Vector-sets), and how Convex.jl reformulates your problem. There could be mathematically equivalent formulations that are faster, and in some cases it could be that Convex's automatic reformulations can be improved.
```